finagle-base-http/src/main/scala/com/twitter/finagle/http/TraceInfo.scala

```diff
@@ -12,6 +12,47 @@ private object TraceInfo {
   }
 
   def letTraceIdFromRequestHeaders[R](request: Request)(f: => R): R = {
+    // rather than rewrite all this to handle more I'm just shimming
+    // in a check for the new trace b3 header and if it exists I'm
+    // extracting out its components to be handled by the existing
+    // code.
+    if (request.headerMap.contains(Header.TraceContext)) {
+      def handleOne(headers: HeaderMap,value: String): Unit =
+        value match {
+          case "d" => headers.set(Header.Flags,"1")
+          case "1" => headers.set(Header.Sampled,"1")
+          case _ => ()
+        }
+      def handleTwo(headers: HeaderMap, a: String,b: String): Unit = {
+        headers.set(Header.TraceId,a)
+        headers.set(Header.SpanId,b)
+      }
+
+      val _ = request.headerMap.get(Header.TraceContext).map(_.split("-").toList) match {
```

Comment:
I think we can omit the `val _ =` bit.  and instead of keeping it in Option, how about we make it a Seq directly, like:

```
request.headerMap.get(Header.TraceContext).toSeq.flatMap(_.split("-")) match { ...
```